src/plugins/controls/public/index.ts

```diff
@@ -9,56 +9,39 @@
 
 import { ControlsPlugin } from './plugin';
 
-export type {
-  ControlGroupApi,
-  ControlGroupRuntimeState,
-  ControlGroupSerializedState,
-  ControlStateTransform,
-  ControlPanelState,
-} from './react_controls/control_group/types';
 export {
   controlGroupStateBuilder,
   type ControlGroupStateBuilder,
 } from './react_controls/control_group/utils/control_group_state_builder';
 
+export type { ControlGroupApi, ControlStateTransform } from './react_controls/control_group/types';
+
 export type {
   DataControlApi,
-  DefaultDataControlState,
   DataControlFactory,
   DataControlServices,
 } from './react_controls/controls/data_controls/types';
-export type { OptionsListControlState } from './react_controls/controls/data_controls/options_list_control/types';
-
-export { ACTION_EDIT_CONTROL } from './react_controls/actions/edit_control_action/edit_control_action';
 
 export {
-  ACTION_DELETE_CONTROL,
   ControlGroupRenderer,
-  type ControlGroupRendererProps,
   type ControlGroupRendererApi,
   type ControlGroupCreationOptions,
-} from './control_group';
-
-export type { ControlWidth, ControlStyle } from '../common/types';
-
-/**
- * TODO: remove all exports below this when control group embeddable is removed
- */
+  type ControlGroupRendererProps,
+} from './react_controls/external_api';
 
 export {
   CONTROL_GROUP_TYPE,
   OPTIONS_LIST_CONTROL,
   RANGE_SLIDER_CONTROL,
   TIME_SLIDER_CONTROL,
 } from '../common';
-
-export {
-  type ControlGroupContainer,
-  ControlGroupContainerFactory,
-  type ControlGroupInput,
-  type ControlGroupOutput,
-  controlGroupInputBuilder,
-} from './control_group';
+export type {
+  ControlGroupRuntimeState,
+  ControlPanelState,
+  ControlPanelsState,
+  DefaultDataControlState,
+} from '../common';
```

Comment:
I've had to export some stuff from `common` to keep the imports consistent for external uses - I kept these minimal though (i.e. only included the types that are used externally), since any internal `controls` code should just import from `common` when necessary.